xml/ha_qdevice-qnetd.xml

```diff
@@ -0,0 +1,508 @@
+                        for clusters with even number of nodes. If the cluster splits
+                        into two similar partitions, it provides one vote to one of
+                        the partitions based on results of any heuristics checks and
+                        other factors.
+                     </para>
+                  </listitem>
+                  <listitem>
+                     <para>
+                        For LMS (<quote>last man standing</quote>), it allows the only
+                        remaining node that can see &qnet; server to get the votes.
+                        So this algorithm is useful when the cluster with only one
+                        active node should remain quorate.
+                     </para>
+                  </listitem>
+               </itemizedlist>
+            </listitem>
+         </varlistentry>
+         <varlistentry>
+            <term>Heuristics</term>
+            <listitem>
+               <para>
+                  &qdevice; supports a set of commands executed locally on startup,
+                  cluster membership change, successful connect to <systemitem
+                     class="daemon">corosync-qnetd</systemitem>, or optionally at
+                  regular times.
+               </para>
+               <para>
+                  When all commands were executed successfully, the heuristics have
+                  passed, otherwise they have failed. The heuristic result is
+                  sent to <systemitem class="daemon">corosync-qnetd</systemitem>
+                  and there it's used in calculations to determine which partition
+                  should be quorate.
+               </para>
+            </listitem>
+         </varlistentry>
+         <varlistentry>
+            <term>Tiebreaker</term>
+            <listitem>
+               <para>
+                  It is used as a fallback if the cluster partitions are completely
+                  equal even with the same heuristics results. It can be configured
+                  to be lowest, higest or specific nodeid.
+               </para>
+            </listitem>
+         </varlistentry>
+      </variablelist>
+   </sect1>
+
+   <sect1 xml:id="sec-ha-qdevice-require">
+      <title>Requirements and Prerequisites</title>
+      <para>
+         Before you start using &qdevice; and &qnet;, make sure the following
+         conditions are fullfilled:
+      </para>
+      <itemizedlist>
+         <listitem>
+            <remark>toms 2020-05-14: See also bsc#1171681</remark>
+            <para>
+               An additional machine which will become the &qnet; server with
+               &sls; and <package>corosync-qnetd</package> package installed.
+            </para>
+         </listitem>
+         <listitem>
+            <para>
+               A network address used by the &qnet; server which can be neutrally
+               reached by all cluster nodes.
```

Comment:
Not sure if mentioned elsewhere, it is mandatory for the production environment to use different individual network interface than the one for corosync. Otherwise, the qnet will not work as expected in the split-brain situation when corosync facing networking failure.

Reply:
> it is mandatory for the production environment to use different individual network interface than the one for corosync

Yes, I'm working on it

Reply:
I'd say it'd be better for corosync-qdevice to reach corosync-qnetd  from a different network interface than the one corosync uses. But why is it mandatory? A split-brain doesn't necessarily mean neither of the nodes could reach qnetd server.

Reply:
Found the "recommendation":

"It is recommended that a quorum device be run on a different physical network at the same site as the cluster that uses the quorum device. Ideally, the quorum device host should be in a separate rack than the main cluster, or at least on a separate PSU and not on the same network segment as the corosync ring or rings."

from here:
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_high_availability_clusters/assembly_configuring-cluster-quorum-configuring-and-managing-high-availability-clusters#proc_configuring-quorum-device-configuring-quorum-devices

There's phrasing issues and confusing usage of terms though. If we are to give the similar recommendation, it could be something like:

"It is recommended for corosync-qdevice to reach  the corosync-qnetd server through a different physical network than the one that corosync uses. Ideally, the corosync-qnetd server  should be in a separate rack than the main cluster, or at least on a separate PSU and not in the same network segment as the corosync ring or rings."

Reply:
Thank you for the feedback! :+1: 

I rephrased the first two sentences a bit to matches it with the other items in the list:

```xml
A different physical network than the one that &corosync; uses.
It is recommended for &qdevice; to reach the &qnet; server.
[... rest is the same ...]
```

Reply:
Not sure about the phrasing here :-)

I understand this is under the context "make sure the following conditions are fullfilled", so this is to be listed as one the the conditions... But the truth is this point is a "recommendation".  I guess we probably need to change the context  a little to be something like "Before setting up &qdevice; and &qnet;,   you need to prepare the environment as the following:"

"
- In addition to the cluster nodes, you have a separate machine which will become the &qnet; server which can be neutrally reached by the cluster nodes

-  It's recommended for qdevice to connect to qnet server through a different physical network than the one that &corosync; uses ...
"

Reply:
Thanks Yan, I agree! That makes sense, integrated your suggestion. :+1: